langchain/agents/tools/sql.ts

```diff
@@ -0,0 +1,217 @@
+import sqlite3 from "sqlite3";
+import { Tool, Toolkit } from "./base";
+import { OpenAI } from "../../llms";
+import { LLMChain } from "../../chains";
+import { PromptTemplate } from "../../prompts";
+
+export class SqlDatabase {
+  private db: sqlite3.Database;
+
+  constructor(db: sqlite3.Database) {
+    this.db = db;
```

Comment:
Would it be better to receive the args to instantiate a SQLite3 instance, so that we could enforce readonly mode here?

Reply:
Yup good point, can change

Reply:
To make testing easier, I'll just turn on readonly mode in the constructor but still take in a db in the constructor